lib/action_pack/cloudfront/eu_location_detection.rb

```diff
@@ -0,0 +1,21 @@
+module ActionPack
+  module Cloudfront
+    module EuLocationDetection
+      EU_ISO_3166_ALPHA_2 = /^(AL|AD|AT|BY|BE|BA|BG|HR|CY|CZ|DK|EE|FI
+                              |FR|DE|GI|GR|HU|IS|IE|IM|IT|RS|LV|LI|LT
+                              |LU|MK|MT|MD|MC|ME|NL|NO|PL|PT|RO|RU|SM
+                              |RS|SK|SI|ES|SE|CH|UA|GB|VA|RS)$/ix
+
+      def request_from_eu?
+        country_code = ActionDispatch::Http::Headers.new(request.headers)['CloudFront-Viewer-Country']
```

Comment:
Mind explains why `request.headers['CloudFront-Viewer-Country']` would not work?

Reply:
It does! I'm not sure, maybe I was setting the header incorrectly when I first started or something and ended up with that after some troubleshooting. But now I'm moving back to the simpler way since it works as well.